pkg/providers/instance/instance.go

```diff
@@ -119,6 +122,7 @@ func (p *Provider) Create(ctx context.Context, nodeClass *v1alpha2.AKSNodeClass,
 	instanceTypes = orderInstanceTypesByPrice(instanceTypes, scheduling.NewNodeSelectorRequirementsWithMinValues(nodeClaim.Spec.Requirements...))
 	vm, instanceType, err := p.launchInstance(ctx, nodeClass, nodeClaim, instanceTypes)
 	if err != nil {
+		// Currently, CSE errors will lead to here
```

Comment:
How does this interact with @charliedmcb's proposal for parallel drift?

Reply:
Given we can treat CSE error as other errors that can arrive there, then I don't think it changes the flow.